pkg/availability/git.go

```diff
@@ -0,0 +1,71 @@
+//
+// Copyright 2024 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package availability
+
+import (
+	"context"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/redhat-appstudio/application-service/pkg/github"
+	"github.com/redhat-appstudio/application-service/pkg/metrics"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+// Inspired from https://github.com/konflux-ci/remote-secret/blob/main/pkg/availability/storage_watchdog.go
+
+type AvailabilityWatchdog struct {
+	GitHubTokenClient github.GitHubToken
+}
+
+func (r *AvailabilityWatchdog) Start(ctx context.Context) error {
+	// Check every 20 minutes
+	ticker := time.NewTicker(20 * time.Minute)
+	go func() {
+		for {
+			// make call immediately to avoid waiting for the first tick
+			r.checkAvailability(ctx)
+			select {
+			case <-ticker.C:
+				continue
+			case <-ctx.Done():
+				ticker.Stop()
+				return
+			}
+		}
+	}()
+	return nil
+}
+
+func (r *AvailabilityWatchdog) checkAvailability(ctx context.Context) {
+	log := ctrl.LoggerFrom(ctx)
+	checkGitLabel := prometheus.Labels{"check": "github"}
```

Comment:
Do we need to update the metric specification doc? It HAS
> has_availability{other_label=”sso”} | konflux_up(service=”has”, check=”sso”)

But `check=sso` doesn't make much sense, and what you have here (`github`), makes more sense

Reply:
updated the doc